CMakeLists.txt

```diff
@@ -1,6 +1,17 @@
-cmake_minimum_required(VERSION 3.8)
 project(bn256)
-add_subdirectory(src)
-add_subdirectory(test)
+cmake_minimum_required(VERSION 3.12)
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_EXTENSIONS ON)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+add_compile_options(-Wall)
+
+find_package(Boost REQUIRED)
 
+add_subdirectory(src)
 
+option(BN256_ENABLE_TEST "BN256_ENABLE_TEST" ON)
```

Comment:
Do we need this option? Should test always be enabled? Will the performance be impacted if test is enabled?

Reply:
Our intention was to create a new CI pipeline for this repo where the tests is only run in this repo and we don't need to enable the tests for leap repo so that we don't even need to build Catch2 for building leap.